TFNetworkLayer.py

```diff
@@ -3520,21 +3520,34 @@ def get_out_data_from_opts(cls, name, sources, axes=None, axis=None, keep_dims=F
     axes = cls.get_axes(axis=axes, input_data=x)
     y_shape = list(x.batch_shape)
     out_batch_dim_axis = x.batch_dim_axis
+    out_feature_dim_axis = x.feature_dim_axis
```

Comment:
`feature_dim_axis` should have some special handling. In case that `x.feature_dim_axis_or_unspecified is NotSpecified`, you should not handle it (unless it is in `axes`). See other related code.